lib/internal/process/esm_loader.js

```diff
@@ -69,6 +69,7 @@ async function initializeLoader() {
         await ESMLoader.import(userLoader, pathToFileURL(cwd).href);
     ESMLoader = new Loader();
     ESMLoader.hook(hooks);
+    ESMLoader.runGlobalPreloadCode();
```

Comment:
The code currently runs after `--require` and I added a test to document that. Before loaders ship, we should definitely make a strong call on which way it should go. Since `--require` code can use `import()`, it seems like the correct thing would be delay it after loader init is done. But that would mean that it runs after the first tick, potentially. Depends on how loader init works in the end. Using a worker thread *may* make it easier to prevent a main thread tick until the loader init is done and the preload code has been determined.